### ISSUE TITLE
Pixelfed nixos module can't serve images

### ISSUE THREAD

**Issue by wlcx** [NixOS/nixpkgs#235147]
### Describe the bug
The Pixelfed nixos module results in a configuration where the web server is unable to read the uploaded user images.

- Media files are uploaded as the pixelfed user, with file mode 0700
- Nginx is then unable to serve those files - e.g 
  ```
  May 31 08:47:57 pixelfed nginx[456]: 2023/05/31 08:47:57 [crit] 456#456: *123 stat() "/nix/store/7bm468m2n4f0pljngwv6ad58283m6l6a-pixelfed-pixelfed/public/storage/m/_v2/568711314458775553/c537ce87c-f5971d/SQE7yPyNuyOo/K3yL7HIA4HazNkCmvJENwANOay7klGFqM6SJFUnZ.jpg" failed (13: Permission denied), client: 192.168.101.2, server: <snip>, request: "GET /storage/m/_v2/568711314458775553/c537ce87c-f5971d/SQE7yPyNuyOo/K3yL7HIA4HazNkCmvJENwANOay7klGFqM6SJFUnZ.jpg HTTP/1.1", host: "<snip>"
  ```
- This comes up in issues on the pixelfed repo, e.g https://github.com/pixelfed/pixelfed/issues/3801 - the common solution seems to be to run the pixelfed php fpm and the web server as the same user.

I am running Pixelfed in a nixos container, but I don't think this should affect anything?

### Steps To Reproduce
1. Configure a pixelfed instance
2. Upload a post
3. Try to view that post (federated or locally)

### Notify maintainers

@RaitoBezarius

**Comment by RaitoBezarius** [comment]
We need to add NGINX to the Pixelfed group I guess.

**Comment by wlcx** [comment]
> We need to add NGINX to the Pixelfed group I guess.

It already is by default, the issue is that files/directories get created with no group read permission. But as per above comment, looks like a version bump might fix this :)

**Comment by wlcx** [comment]
Yeah I think I'm seeing this too.

**Comment by jenselofsson** [comment]
I tried adding "FILESYSTEM_DRIVER" = "local" to the pixelfed settings, as detailed in https://docs.pixelfed.org/technical-documentation/config/#filesystem_driver

But it had no effect at all. It's also a bit confusing since "local" is the default, but the path where the images are stored makes it seem like "public" is used.

**Comment by kivikakk** [comment]
Can confirm still reproduces on 0.11.11 (i.e. as of nixos-23.11). `storage` is 0750, as is `storage/m`, but `storage/m/_v2` is 0700, as are its subdirectories and files.

**Comment by RaitoBezarius** [comment]
Argh, I missed the reopening. I will take a look again when I can.

**Comment by riotbib** [comment]
May confirm, that this still reproduces on v0.11.13.

I am new to hosting Pixelfed, so the following might not come as a surprise for some, but restarting `pixelfed-data-setup.service` manually solves the problem.

I have a kinda untouched Pixelfed instance running and can help debugging with that one. Do let me know, in case you'd need or want help, @RaitoBezarius.

**Comment by riotbib** [comment]
For your information, implementing [this patch](https://git.hubrecht.ovh/hubrecht/nixos/src/commit/411ce0386c475eb569381b6a9f719ec49e4fc04e/machines/dell-r730/pixelfed/default.nix#L7-L9) by @Tom-Hubrecht fixed it for me.

**Comment by Niols** [comment]
This patch works for me too.

However, applying this patch and adding nginx to the `pixelfed` group amounts to saying that everything that Pixelfed flags as “private” should be available to the whole web.

Instead, I feel Pixelfed private medias should be served by Pixelfed after proper authentication, and not by nginx. Public medias could still be served by nginx. Isn't it then an upstream issue that Pixelfed flags as private medias that are in fact public?

**Comment by Radvendii** [comment]
It seems from [these installation docs](https://pixelfed.github.io/docs-next/running-pixelfed/installation.html) that pixelfed assumes it is running under the same user as nginx.
> ```
> ...
> [Service]
> User=http
> ...
> ```
And 
> The example above assumes... that your distribution uses the http user to serve Nginx.

This seems like a Bad Choice, but it does mean that giving group read permissions to those files is not going to be any worse than a typical pixelfed installation (and in fact, is better because pixelfed still cannot read everything nginx can)

**Comment by flyfloh** [comment]
> May confirm, that this still reproduces on v0.11.13.
> 
> I am new to hosting Pixelfed, so the following might not come as a surprise for some, but restarting `pixelfed-data-setup.service` manually solves the problem.

I have the same behavior on v.012.3 :-(

**Comment by h3ndrik** [comment]
And still an issue with 0.12.3 and 0.12.4

The patch works. I think we should add that to the package. Are we making anything worse, if the recommended default config also grants nginx all permissions, by running it as the same user?

**Comment by Niols** [comment]
By adding `nginx` to the `pixelfed` group, we allow easy escalation from `nginx` to the Pixelfed instance, but that's basically by design of Pixelfed, so there isn't much that we can do. By running them as the same user, however, we also allow escalation from Pixelfed to `nginx`, and I think we should avoid that.

**Comment by h3ndrik** [comment]
> [...] By running them as the same user, however, we also allow escalation from Pixelfed to `nginx`, and I think we should avoid that.

Agreed. And we don't want to do that. Pixelfed runs fine with just the patch from above https://github.com/NixOS/nixpkgs/issues/235147#issuecomment-2266176071 and without running it as the same user.

**Comment by riotbib** [comment]
I updated Pixelfed to v1.12.5, it threw errors because the patch by @Tom-Hubrecht could not be applied. Removing the override for the patch made it obvious the bug described in this issue is no longer existing. Uploading an image after rebuild worked flawlessly.

Could someone who was also applying this patch confirm? I guess it's because of [this change to upstream](https://github.com/pixelfed/pixelfed/pull/5765). Thanks!

**Comment by phanirithvij** [comment]
Assuming this has been solved from the recent comments on this thread, re-open if it is still an issue.
